### PR TITLE
DAOS-11467 ucx: Update UCX version to 1.13.

### PR DESCRIPTION
**Pull request by jgmoore-or**
An update to UCX 1.13 is required to enable needed changes in the Mercury UCX adapter.

### CODE REVIEW

File: ucx.spec [hunk 1]
```diff
@@ -91,11 +98,17 @@ The acronym UCX stands for "Unified Communication X".
 Requires: %{name}%{?_isa} = %{version}-%{release}
 Summary: Header files required for developing with UCX
 
+%if (0%{?suse_version} > 0)
```
Comment (jolivier23) [inline ucx.spec:101]: @brianjmurrell please advise if this is ok or hacky.  It appears to work for the build.  At present, we've been using the system version (1.11) on leap

**Review by jolivier23 — DISMISSED** [review]

File: ucx.spec [hunk 2]
```diff
@@ -19,14 +19,21 @@
 %bcond_with    xpmem
 %bcond_with    vfs
 
+%global major 1
+%global minor 13
+%global bugrelease 0
+#%%global prerelease
+
+%global dl_version %{major}.%{minor}.%{bugrelease}
```
Comment (brianjmurrell) [inline ucx.spec:27]: Nice.  Thanks for updating this with this standard.

File: ucx.spec [hunk 3]
```diff
@@ -106,19 +119,19 @@ Provides header files and examples for developing with UCX.
...
+           --without-go \
+          %_enable_arg cma cma \
+          %_with_arg cuda cuda \
+          %_with_arg gdrcopy gdrcopy \
+          %_with_arg ib verbs \
+          %_with_arg ib_cm cm \
+          %_with_arg knem knem \
+          %_with_arg rdmacm rdmacm \
+          %_with_arg rocm rocm \
+          %_with_arg xpmem xpmem \
```
Comment (brianjmurrell) [inline ucx.spec]: Inadvertent removal of leading whitespace?  These lines should line up with the lines above them.

File: ucx.spec [hunk 4]
```diff
@@ -106,19 +119,19 @@ Provides header files and examples for developing with UCX.
...
            %_with_arg vfs fuse3 \
-           %_with_arg ugni ugni \
-           %{?configure_options}
+          %_with_arg ugni ugni \
+          %{?configure_options}
```
Comment (brianjmurrell) [inline ucx.spec]: Ditto.

File: ucx.spec (hunk 1 above)
Comment (brianjmurrell) [reply ucx.spec:101]: Hrm.  I'm on the fence about this.  When we build a package for a platform, we really should be trying to build apples-to-apples packages.

In this case however, at runtime UCX would be coming from MLNX, not SUSE, and not the artifacts of this PR.  I'm not sure how UCX is packaged for MOFED on Leap 15 though.  I would not be at all surprised to learn that MLNX are not building apples-to-apples packages for SUSE distros either.

I guess the ultimate question is does building like this on Leap 15 allow the resulting `libuc{s,t,p}-devel` packages to then be used to build mercury and then allow that mercury to run with MOFED UCX on Leap 15?

If we are going to be claiming to support UCX with SUSE and MOFED, the above really needs to be verified.

**Comment by jgmoore-or** [comment]
I checked, and the UCX install from MOFED on Leap is no different than for CentOS or Rocky.  This PR is required to enable the change to Mercury UCX adapter in daos-stack/mercury.

**Comment by jgmoore-or** [comment]
I installed the leap RPM from this PR on Leap 15.3 without any issues.  Re-imaged with leap 15.4 (has MOFED 5.6 and UCX 1.13) and was able to install the RPM without dependency issues.

**Review by jolivier23 — DISMISSED** [review]

**Comment by jolivier23** [comment]
@brianjmurrell are you ok with this now that it's been verified that it doesn't cause install issues with mofed?

**Review by brianjmurrell — DISMISSED** [review]

**Review by jolivier23 — APPROVED** [review]

**Review by frostedcmos — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**